### ISSUE TITLE
Breakpoints not hit on the fsprovider sample

### ISSUE THREAD

**Issue by LiamLamb** [microsoft/vscode-extension-samples#1134]
### Extension sample

fsprovider-sample

### VS Code version

1.95.3

### What went wrong?

I have been having issues getting debugging to work on the `fsprovider` sample. 

When executing F5 to launch the extension, I would expect a breakpoint in the activation method to be called, but it never is. 

Here is some information about the environment.

Code information:
```
Version: 1.95.3 (Universal)
Commit: f1a4fb101478ce6ec82fe9627c43efbf9e98c813
Date: 2024-11-13T14:50:04.152Z
Electron: 32.2.1
ElectronBuildId: 10427718
Chromium: 128.0.6613.186
Node.js: 20.18.0
V8: 12.8.374.38-electron.0
OS: Darwin x64 22.6.0
```

Commit used: [5ddd30fc052](https://github.com/microsoft/vscode-extension-samples/commit/5ddd30fc052e03bbec52e5d84627eaa543fb0de8)

**Comment by connor4312** [comment]
Is the extension actually getting activated?

**Comment by LiamLamb** [comment]
Yes, the extension is being activated as expected and works perfectly well otherwise. 

Having played around with it a bit more, I am not convinced this is an issued with the samples. I've noticed I can consistently reproduce the issue when I am developing on battery. When I am plugged into a power source, the debugging works as expected, which is a _tad_ confusing. 

I also tried on a windows machine and couldn't replicate it there. 

Unless you can think of anything which is worth investigating in this area, I have something that works so we could probably close this issue.

**Comment by connor4312** [comment]
This works for me, if please grab a log file using these instructions if you can still reproduce it:

> If you're able to, add `"trace": true` to your `launch.json` and reproduce the issue. The location of the log file on your disk will be written to the Debug Console. Share that with us.

**Comment by LiamLamb** [comment]
Sure, here you go:

[vscode-debugadapter-7d188ed1.json.gz](https://github.com/user-attachments/files/18132578/vscode-debugadapter-7d188ed1.json.gz)

**Comment by connor4312** [comment]
It looks like VS Code never notified the debugger that the new window and extension host is opened. I'm not sure what the deal there is, or why it would work when plugged in and not on battery 🤔
